src/MomentumPyClient/ui.py

```diff
@@ -34,14 +30,43 @@ def get_container_definitions(self):
             "brown",
             "cyan",
             "gray",
+            "Cyan",
+            "Magenta",
+            "Teal",
+            "Pink",
+            "Lime",
+            "Lavender",
+            "Beige",
+            "Maroon",
+            "Mint",
+            "Peach",
+            "Navy",
+            "Olive",
+            "Coral",
         ]
-        i = 0
-        for container in self.containers:
-            container["color"] = color_names[i]
-            i += 1
-            if i >= len(color_names):
-                i = 0
-        return self.containers
+        self.color_dict = {}
+        self.color_index = 0
+        containers = self.ws.get_container_definitions()
+        # Create a dictioary to lookup the color
+        for container in containers:
+            #            self.get_container_color(container["Name"])
+            self.get_container_color(container["InventoryTemplateName"])
+
+    def set_container_colors(self, color_dict: dict):
+        self.color_dict = color_dict
+
+    def get_container_color(self, container_name):
+        if container_name is None:
+            return "blue"
+        if container_name in self.color_dict:
```

Comment:
```suggestion
        if container_name not in self.color_dict:
            self.color_dict[container_name] = next(self.color_provider)
        return self.color_dict[container_name]
```
could replace lines 61-69 (if we initialize `self.color_provider` in `__init__`

Reply:
Can we have this together with the opportunity to specify your own favorite colurs like : 

colorDict = {
        "T_OrangeGTest": "orange",
        "T_384_Assay": "yellow",
        "I_384_Assay": "yellow",
        "I_SaltTitration_Source": "red",
        "T_SaltTitration_Source": "red",
        "T_AC_SINS_Destination": "gold",
        "I_AC_SINS_Destination": "gold",
        "T_Displacement_Assay": "green",
        "I_Displacement_Assay": "green",
        "T_Displacement_Source": "purple",
        "I_Displacement_Source": "purple",
        "I_DLS_Stability": "black",
        "T_DLS_Stability": "Black",
    }
If you have time can you implement the changes suggested ?